### PR TITLE
Updated opentelemetry to 0.15

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Fixes #572

## Proposed Changes

- Updated opentelemetry to 0.15, removed guava exclusion (opentelemetry 0.15 doesn't depend on guava anymore), also changed some deprecated code to ensure next bump goes fine

### CODE REVIEW

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java [hunk 1]
```diff
@@ -57,46 +60,42 @@ public static void setup(final TracingConfig tracingConfig) {
...
 
     if (tracingConfig.getBackend().equals(Backend.ZIPKIN)) {
-
       logger.debug("Add Zipkin processor");
-
-      tracerManagement.addSpanProcessor(
+      tracerProviderBuilder.addSpanProcessor(
         BatchSpanProcessor
           .builder(zipkinExporter(tracingConfig))
-          .setExportOnlySampled(true)
```
Comment (pierDipi) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java:74]: Can't be set anymore?

**Review by pierDipi — COMMENTED** [review]
Can you open an issue for migrating this? https://github.com/knative-sandbox/eventing-kafka-broker/blob/482f1c3baffbab9dabfc6d80b4c306f4713b1c07/data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/TracingSpan.java#L41 since: 
> The items in the io.opentelemetry.semconv.trace.attributes.SemanticAttributes which were previously generated form the Resource semantic conventions have been deprecated. Please use the ones in the new io.opentelemetry.semconv.resource.attributes.ResourceAttributes class.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java (hunk 1 above)
Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java:74]: It seems useful.

Comment (slinkydeveloper) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java:74]: Now you need to set it like this: https://github.com/knative-sandbox/eventing-kafka-broker/pull/587/files#diff-5251f57656ad46ff5e6738a59c628075ea3274249446aca9807bae90f77606b3R65

Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java:74]: That is the same as https://github.com/knative-sandbox/eventing-kafka-broker/pull/587/files#diff-5251f57656ad46ff5e6738a59c628075ea3274249446aca9807bae90f77606b3L63, this sets whether we export an unsampled trace or not based on that sampling ratio.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java [hunk 2]
```diff
@@ -49,9 +47,7 @@
 
   private static final Logger logger = LoggerFactory.getLogger(Tracing.class);
 
-  private static final AtomicReference<SdkTracerProvider> trackerProvider = new AtomicReference<>();
-
-  public static void setup(final TracingConfig tracingConfig) {
+  public static SdkTracerProvider setup(final TracingConfig tracingConfig) {
```
Comment (pierDipi) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java:50]: Thanks for cleaning this up!

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java [hunk 3]
```diff
@@ -79,15 +79,15 @@ public static void main(final String[] args) throws IOException {
 
     final var env = new DispatcherEnv(System::getenv);
 
-    Tracing.setup(TracingConfig.fromDir(env.getConfigTracingPath()));
+    SdkTracerProvider sdkTracerProvider = Tracing.setup(TracingConfig.fromDir(env.getConfigTracingPath()));
```
Comment (pierDipi) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:82]: nit
```suggestion
    final SdkTracerProvider sdkTracerProvider = Tracing.setup(TracingConfig.fromDir(env.getConfigTracingPath()));
```

File: data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/Main.java [hunk 4]
```diff
@@ -77,7 +77,7 @@
   public static void main(final String[] args) throws IOException {
     final var env = new ReceiverEnv(System::getenv);
 
-    Tracing.setup(TracingConfig.fromDir(env.getConfigTracingPath()));
+    SdkTracerProvider sdkTracerProvider = Tracing.setup(TracingConfig.fromDir(env.getConfigTracingPath()));
```
Comment (pierDipi) [inline data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/Main.java:80]: nit:
```suggestion
    final SdkTracerProvider sdkTracerProvider = Tracing.setup(TracingConfig.fromDir(env.getConfigTracingPath()));
```

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java (hunk 1 above)
Comment (slinkydeveloper) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java:74]: From my understanding of the release notes, now this is just done by the ratio sampler

**Comment by slinkydeveloper** [comment]
> Can you open an issue for migrating this? 

I'll do it right now

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java (hunk 1 above)
Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/Tracing.java:74]: Ok, thanks!
/lgtm

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java [hunk 5]
```diff
@@ -155,6 +155,7 @@ public static void main(final String[] args) throws IOException {
       logger.error("Failed during filesystem watch", ex);
 
       Shutdown.closeSync(vertx).run();
+      sdkTracerProvider.close();
```
Comment (pierDipi) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:158]: I think we should close this even if there are no exception.

**Review by pierDipi — COMMENTED** [review]
/hold

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java (hunk 5 above)
Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:158]: Some line above we add a shutdown hook.

Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:158]: Same question for the receiver.

Comment (slinkydeveloper) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java]: I did a little fancy change to Shutdown hook, check it out

**Review by pierDipi — COMMENTED** [review]
/lgtm
/retest

**Pull request merged**